src/actions/walletConnectActions.js

```diff
@@ -215,8 +217,18 @@ const subscribeToSessionRequestEvent = (connector: Connector) => {
 
         return;
       }
-      dispatch(walletConnectSessionReceived());
 
+      if (!shouldAllowSession(peerMeta.url)) {
+        Toast.show({
+          type: 'warning',
+          title: 'Cannot connect',
+          message: 'We are sorry. We do not support this application.',
```

Comment:
Maybe it's better to say that application you are trying to connect doesn't support Smart accounts?